nixos/modules/image/repart-verity-store.nix

```diff
@@ -0,0 +1,209 @@
+# opinionated module that can be used to build nixos images with
+# a dm-verity protected nix store
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}:
+let
+  cfg = config.image.repart.verityStore;
+
+  verityMatchKey = "store";
+
+  # TODO: make these and other arch mappings available from systemd-lib for example
+  partitionTypes = {
+    usr =
+      {
+        "x86_64" = "usr-x86-64";
+        "arm64" = "usr-arm64";
+      }
+      ."${pkgs.stdenv.hostPlatform.linuxArch}";
+
+    usr-verity =
+      {
+        "x86_64" = "usr-x86-64-verity";
+        "arm64" = "usr-arm64-verity";
+      }
+      ."${pkgs.stdenv.hostPlatform.linuxArch}";
+  };
+
+  verityHashCheck =
+    pkgs.buildPackages.writers.writePython3Bin "assert_uki_repart_match.py"
+      {
+        flakeIgnore = [ "E501" ]; # ignores PEP8's line length limit of 79 (black defaults to 88 characters)
+      }
+      (
+        builtins.replaceStrings [ "@NIX_STORE_VERITY@" ] [
+          partitionTypes.usr-verity
+        ] (builtins.readFile ./assert_uki_repart_match.py)
+      );
+in
+{
+  options.image.repart.verityStore = {
+    enable = lib.mkEnableOption "enable building images with a dm-verity protected store";
+
+    ukiPath = lib.mkOption {
+      type = lib.types.str;
+      default = "/EFI/Linux/${config.system.boot.loader.ukiFile}";
```

Comment:
This may result in an infinite recursion loop because of the two staged builds now (?).

Reply:
More specifically, if you have something like:

```nix
"/EFI/Linux/${config.system.boot.loader.ukiFile}".source =
${config.system.build.uki}/${config.system.boot.loader.ukiFile}";
```

in the contents of your ESP.

I think the UKI is gonna loop.

Reply:
Hmm, is this actually something that needs to be considered?

The injection of the generated UKI is done by the module in the final image build, the user doesn't need to specify it manually in `contents` of the ESP repart config.

And in case the target path on the ESP should be different from `/EFI/Linux/${...ukiFile}` the `ukiPath` option would be the way to handle that (as is done in the test now to boot the UKI directly rather than going through systemd-boot).

Reply:
_Ah_ !

Reply:
OK, let me test this, sorry.